Source/WebCore/animation/AnimationTimelinesController.cpp

```diff
@@ -714,7 +716,10 @@ void AnimationTimelinesController::unregisterNamedTimelinesAssociatedWithElement
     for (auto& entry : m_nameToTimelineMap) {
         auto& timelines = entry.value;
         timelines.removeAllMatching([&] (const auto& timeline) {
-            return originatingElement(timeline) == styleable;
+            if (originatingElement(timeline) != styleable)
+                return false;
+            m_removedTimelines.add(timeline.get());
```

Comment:
I don't think a block that is designed as a boolean test should ever have side effects.

This would be better as a filter function that gathered a list of to-be-removed timelines, then removed them.

Reply:
Agreed, I've made the necessary change in the commit.